src/main/java/de/tum/in/www1/artemis/repository/metrics/ExerciseMetricsRepository.java

```diff
@@ -107,7 +107,7 @@ CASE WHEN TYPE(e) = ProgrammingExercise THEN TREAT(e AS ProgrammingExercise).all
             WHERE e.id IN :exerciseIds
                 AND s.submitted = TRUE
                 AND (p.student.id = :userId OR u.id = :userId)
-            GROUP BY p.id
+            GROUP BY e.id, p.id
```

Comment:
Why do you group by participationId as well?

Reply:
Good point, technically I could just group by exercise id now as far as I can see. I will quickly test it out

Reply:
I suggest we add a comment, but this is due to Postgres, otherwise the jdbc driver throws an exception

Reply:
I reverted my commit :| I think we need it because of this: "It takes into account both individual and team participations."

Reply:
But still I think you could just remove the `p.id` and group by `e.id`